src/SmartComponents/modules/reducers.js

```diff
@@ -88,6 +89,70 @@ function selectedSystems(selectedIds, selectedSystem) {
     return selectedIds;
 }
 
+function convertFactsToCSV(data, systems) {
+    let result;
+    let keys;
+    let columnDelimiter;
+    let lineDelimiter;
+
+    if (data === null || !data.length) {
+        return null;
+    }
+
+    columnDelimiter = data.columnDelimiter || ',';
+    lineDelimiter = data.lineDelimiter || '\n';
+
+    result = 'Fact,State,';
+    result += systems.join(columnDelimiter);
+    result += lineDelimiter;
```

Comment:
This builds up the 'headers', is that correct? If so, it would help my readability to have this as a separate variable and then combine them at the end, e.g.  headers + result